pandas/tests/groupby/test_function.py

```diff
@@ -555,6 +555,83 @@ def test_idxmin_idxmax_axis1():
         gb2.idxmax(axis=1)
 
 
+@pytest.mark.parametrize("numeric_only", [True, False, None])
+def test_axis1_numeric_only(groupby_func, numeric_only):
+    if groupby_func in ("idxmax", "idxmin"):
+        pytest.skip("idxmax and idx_min tested in test_idxmin_idxmax_axis1")
+    if groupby_func in ("mad", "tshift"):
+        pytest.skip("mad and tshift are deprecated")
+    if groupby_func in ("corrwith", "skew"):
+        msg = "GH#47723 groupby.corrwith and skew do not correctly implement axis=1"
+        pytest.skip(msg)
+
+    df = DataFrame(np.random.randn(10, 4), columns=["A", "B", "C", "D"])
+    df["E"] = "x"
+    groups = [1, 2, 3, 1, 2, 3, 1, 2, 3, 4]
+    gb = df.groupby(groups)
+    method = getattr(gb, groupby_func)
+    args = (0,) if groupby_func == "fillna" else ()
+    kwargs = {"axis": 1}
+    if numeric_only is not None:
+        # when numeric_only is None we don't pass any argument
+        kwargs["numeric_only"] = numeric_only
+
+    # Functions without numeric_only and axis args
+    no_args = ("cumprod", "cumsum", "diff", "fillna", "pct_change", "rank", "shift")
+    # Functions with axis args
+    has_axis = (
+        "cumprod",
+        "cumsum",
+        "diff",
+        "pct_change",
+        "rank",
+        "shift",
+        "cummax",
+        "cummin",
+        "idxmin",
+        "idxmax",
+        "fillna",
+    )
+    if numeric_only is not None and groupby_func in no_args:
+        try:
+            method(*args, **kwargs)
+            assert False, f"axis=1 succeeds for {groupby_func}"
+        except TypeError as err:
```

Comment:
Could `pytest.raises` be used for these `TypeError` fallbacks? It's not entirely clear what methods should hit these.

Reply:
Doh - I write these in a notebook and keep forgetting to convert them over to pytest; thanks.

> It's not entirely clear what methods should hit these.

This is for functions that don't accept numeric_only. I was thinking this was clear from checking the error message "got an unexpected keyword argument 'numeric_only'", but maybe it could be improved?

Reply:
Got it. It's a lot clearer from your latest commits, thanks!